### DISCUSSION TITLE
Ability replace / ignore sub-dependencies

### DISCUSSION THREAD

**Discussion by VaZark** [python-poetry/poetry#8033] [Q&A]
In one of my projects, a dependency uses `psycopg2` whereas the project already uses a `psycopg2-binary`.

Besides the redundancy, there are cases when the `psycopg2` package does not build on some machines (Macs) and the recommended resolution is to use `psycopg2-binary`. However, we're unable to find a definitive way to override the subdependency without forking the project (we've raised the issue with the dependency author as well)

It'd be great if we could override/ignore a dependency via poetry as well

**Comment by clintonroy** [comment]
Poetry aims to build wheels that work within the overall wheel system; wheels do not have a way of overriding dependencies, hence poetry won't do anything special in that regard.

There is talk of adding virtual style dependencies in the works, upstream, that might solve these issues, eventually.
